isis/tests/HistoryTests.cpp

```diff
@@ -0,0 +1,48 @@
+#include "Fixtures.h"
+#include "History.h"
+
+#include "gmock/gmock.h"
+
+using namespace Isis;
+
+TEST(HistoryTests, HistoryTestsDefaultConstructor) {
+  History history;
+
+  Pvl historyPvl  = history.ReturnHist();
+  EXPECT_EQ(historyPvl.groups(), 0);
+}
+
+TEST_F(HistoryBlob, HistoryTestsFromBlob) {
+  History readHistory(*historyBlob);
+
+  Pvl historyPvl = readHistory.ReturnHist();
+  ASSERT_TRUE(historyPvl.hasObject("mroctx2isis"));
+  EXPECT_TRUE(historyPvl.findObject("mroctx2isis").hasGroup("UserParameters"));
+}
+
+TEST_F(HistoryBlob, HistoryTestsAddEntry) {
+  History history;
+
+  history.AddEntry(historyPvl);
+
+  Pvl newHistoryPvl = history.ReturnHist();
+  ASSERT_TRUE(newHistoryPvl.hasObject("mroctx2isis"));
+  EXPECT_TRUE(newHistoryPvl.findObject("mroctx2isis").hasGroup("UserParameters"));
+}
+
+TEST_F(HistoryBlob, HistoryTeststoBlob) {
+  History history(*historyBlob);
+
+  Blob *blob = history.toBlob();
+
+  std::stringstream os;
+  char *blob_buffer = blob->getBuffer();
+  Pvl newHistoryPvl;
+  for (int i = 0; i < blob->Size(); i++) {
+    os << blob_buffer[i];
+  }
+  os >> newHistoryPvl;
```

Comment:
I wouldn't interrogate the Blob itself, I would just test History -> Blob -> History works correctly.

Reply:
Maybe? That's testing a lot in one test, which is why I thought chunking the test up would be better